### ISSUE TITLE
EmrEtlRunner: explore config.yml validation against schema to produce meaningful errors 

### ISSUE THREAD

**Issue by chuwy** [snowplow/snowplow#2406]
As config.yml keeps growing it becoming harder to configure it right way from first try. And current [error](https://groups.google.com/forum/#!topic/snowplow-user/wNQPA2psbbQ) [messages](https://groups.google.com/forum/#!topic/snowplow-user/UitKVlXlJqM) are not very helpful.
We can use something like [kwalify](http://www.kuwata-lab.com/kwalify/ruby/users-guide.01.html#schema-seq) to produce more meaningful tips.
Also, YAML 1.2 is superset of JSON, may be there's a way to use JSON Schema for it.

**Comment by alexanderdean** [comment]
Yes, this is a good suggestion! We originally planned to use something like kwalify, but I think a more Snowplow-esque course of action would be:

1. Move out the storage targets into their own configuration files as planned (i.e. similar to enrichments)
2. Create a self-describing schema for the remainder
3. Validate that self-describing schema on startup

**Comment by alexanderdean** [comment]
Work has started on this:

* https://github.com/snowplow/snowplow/issues/2991
* https://github.com/snowplow/snowplow/issues/2992
